hedera-node/docs/design/services/smart-contract-service/RedirectContract-description.md

```diff
@@ -0,0 +1,224 @@
+# The "Redirect" Contract Template for proxying calls to Tokens, Accounts, and Scheduled Transactions
+
+The "redirect" contracts (aka proxy contracts) implement the feature where you can call a Hedera
+token or account or scheduled transaction directly at their address and they _act_ as if you made
+the corresponding contract call.
+
+There is one redirect contract which is parameterized as a _template_ with the addresses of the
+HTS, HAS, and HSS system contracts, and with the correct "redirect" method selector.
+
+When a call is made to a token, an account, or a scheduled transaction, the redirect template is
+instantiated with the correct system contract address and selector, and then executed.  It wraps
+the calldata (provided by the calling contract or transaction) and delegate-calls the actual
```

Comment:
this is the one exception where a delegate call is allowed to a system contract function

Reply:
I will mention that ....